src/app/forms/components/suggestive-input/suggestive-input.component.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormGroup, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { MultiOption } from '@forms/models/options.model';
+import { CustomFormControl, FormNodeTypes } from '@forms/services/dynamic-form.types';
+import { Observable, of } from 'rxjs';
+import { FieldErrorMessageComponent } from '../field-error-message/field-error-message.component';
+import { SuggestiveInputComponent } from './suggestive-input.component';
+
+@Component({
+  selector: 'app-host-component',
+  template: `<form [formGroup]="form">
+    <app-suggestive-input name="foo" formControlName="foo" [options$]="options$"></app-suggestive-input>
+  </form> `,
+  styles: []
+})
+class HostComponent {
+  form = new FormGroup({
+    foo: new CustomFormControl({ name: 'foo', type: FormNodeTypes.CONTROL, children: [] }, '')
+  });
+  options$: Observable<MultiOption[]> = of([
+    { label: 'Banana', value: 'banana' },
+    { label: 'Apple', value: 'apple' }
+  ]);
+}
+
+describe('SuggestiveInputComponent', () => {
+  let component: HostComponent;
+  let fixture: ComponentFixture<HostComponent>;
+  let suggestiveInput: SuggestiveInputComponent;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [HostComponent, SuggestiveInputComponent, FieldErrorMessageComponent],
+      imports: [FormsModule, ReactiveFormsModule]
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(HostComponent);
+    component = fixture.componentInstance;
+    suggestiveInput = fixture.debugElement.query(By.directive(SuggestiveInputComponent)).componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe(SuggestiveInputComponent.prototype.handleChangeForOption.name, () => {
+    it('should find matching option and patch value', async () => {
+      await suggestiveInput.handleChangeForOption('Banana');
+      expect(component.form.get('foo')?.value).toBe('banana');
+    });
+
+    it('should not find matching option and patch `[INVALID_OPTION]`, control should also be invalid', async () => {
+      await suggestiveInput.handleChangeForOption('Orange');
+      const foo = component.form.get('foo');
+      expect(foo?.value).toBe('[INVALID_OPTION]');
+      expect(foo?.invalid).toBeTruthy();
+    });
+
+    it('should not find matching option and patch empty string', async () => {
+      await suggestiveInput.handleChangeForOption('');
+      expect(component.form.get('foo')?.value).toBe('');
+    });
```

Comment:
Maybe some quick unit tests for `get style()`, `findOption()` & `addValidators()`?

Reply:
`findOption()` and `addValidators()` are tested indirectly with present tests. Propagated control values would not be as expected otherwise.

Reply:
Fair enough.

Reply:
I did add one for `get style()` though.